snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -571,13 +571,25 @@ def prime(self, force=False) -> None:  # noqa: C901
         elif is_classic:
             dynamic_linker = self._project_options.get_core_dynamic_linker()
 
-        if dynamic_linker:
+        if dynamic_linker and not self._build_attributes.no_patchelf():
+            logger.warning(
+                'Files in this part are going to be patched to work correctly '
+                'in the environment.\n'
```

Comment:
I suggest: Files in this part are going to be patched to execute correctly in any environment that supports snaps.

Reply:
How about "Files in this part are going to be patched to execute correctly on diverse end-user systems."